### PR TITLE
sunyar-fabric

### PR DESCRIPTION
**Pull request by sunyar-fabric**
## Checklist for Hyperledger Lab Proposal PR

- [x] Read [README.md](https://github.com/hyperledger-labs/hyperledger-labs.github.io/blob/master/README.md) carefully
- [x] Make sure your changes are committed with the proper sign-off
- [x] Use the labs name as the PR title
- [x] Copy the short description of the proposal as the description of the PR
- [x] Remove this checklist section
- [x] Submit your PR and the labs stewards will review your proposal
- [x] Ask your sponsor to confirm sponsorship by stating so in a comment to the PR

### CODE REVIEW

File: sunyar-fabric.md
Comment (tkuhrt) [inline sunyar-fabric.md:4]:
```suggestion
parent: Labs
```

File: sunyar-fabric.md
Comment (tkuhrt) [inline sunyar-fabric.md:7]: The lab name is supposed to be the same as the name of the .md file.
```suggestion
Sunyar Fabric
```

File: sunyar-fabric.md
Comment (tkuhrt) [inline sunyar-fabric.md:9]: This should be a short description of the GitHub repo that would be created for this lab. In other words, describe in a sentence the lab Sunyar Fabric.

File: sunyar-fabric.md
Comment (tkuhrt) [inline sunyar-fabric.md:13]: This should be the scope of what you are intending to create with this lab. As mentioned in the template, please provide enough information for us to understand how this fits with Hyperledger's mission.

File: sunyar-fabric.md
Comment (tkuhrt) [inline sunyar-fabric.md:24]: We need names of the people who will be contributing to this lab. Without that, no one will have write access to commit to the GitHub repo.

File: sunyar-fabric.md
Comment (tkuhrt) [inline sunyar-fabric.md:28]: I received a 404 when attempting to look at this repo in GitHub. Please make the repo public so that we may check things such as license and DCO sign offs. This will be required before we allow for this lab to be approved.

**Review by ZankoMohammadzadeh — APPROVED** [review]

**Comment by sunyar-fabric** [comment]
All of reviews have been done. please check them.

File: sunyar-fabric.md
Comment (tkuhrt) [inline sunyar-fabric.md:10]:
```suggestion
Sunyar is a settlement distributed system based on Hyperledger Fabric. This is a non-profit project for contributing charities to help needy people in a decentralized ecosystem.
```

Comment (tkuhrt) [reply sunyar-fabric.md:10]: Please note that there is another lab focused on charity tracking and Hyperledger Fabric -- [Karma Charity Platform](https://labs.hyperledger.org/labs/karma-charity-platform.html). This might be a good lab to collaborate with for ongoing work.

File: sunyar-fabric.md
Comment (tkuhrt) [inline sunyar-fabric.md:13]:
```suggestion
Sunyar is a settlement distributed system based on Hyperledger Fabric. There are several use cases in banking. Our goal is to achieve an appropriate project on Hyperledger Fabric to solve other challenges and use cases on banking systems.
```

Comment (tkuhrt) [reply sunyar-fabric.md:13]: From the README in the repo linked below, there is the following description:

> Sunyar is a distributed settlement platform, which can be used by charities for managing donation process. This can be done by an open-source platform, which can link to a distributed ledger for charities. The open-source platform helps them for automating their process and the distributed ledger can used for two main reasons: uniqueness of needy and their requirements, and transparent aid for them).

Based on the scope given above, it looks like the intent is to go beyond just the charity aspect. Is this true?

File: sunyar-fabric.md
Comment (sunyar-fabric) [reply sunyar-fabric.md:10]: Sounds great. we are trying to read their project and contribute on required situations.

File: sunyar-fabric.md
Comment (sunyar-fabric) [reply sunyar-fabric.md]: Actually our Lab has focused on distributed settlement to accomplish other use cases in Fintech and banking systems such as Post-trade securities and Cross border payments.

File: sunyar-fabric.md
Comment (sunyar-fabric) [reply sunyar-fabric.md:30]: thank you for your review and effort.
Do you mean amending existing commits of "hyperledger-labs.github.io" repository?

Comment (sunyar-fabric) [reply sunyar-fabric.md:30]: So, I should close this fork, and create another fork?

Comment (tkuhrt) [reply sunyar-fabric.md:30]: For option 1, you would be amending the commits in https://github.com/sunyar-fabric/sunyar. See https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md. A quick solution to signoff last X number of commits.

```
git rebase --signoff @~X
```

For option 2, we would create the new lab repo and then you would manually transfer your code from the existing repo into this newly created lab repo.

Your choice.

**Comment by tkuhrt** [comment]
Please restore the `proposal-template.md` file. This file should not be deleted.

File: sunyar-fabric.md
Comment (tkuhrt) [reply sunyar-fabric.md:30]: I see that there are still some commits that have not been signed off.

Comment (sunyar-fabric) [reply sunyar-fabric.md:30]: I cloned the repository into a directory. 
I committed this command.
$ git rebase -i HEAD~20
then I pushed forced to the repository again.
please check the DCO, commits have been signed off.
thank you

**Review by knagware9 — APPROVED** [review]

**Comment by sunyar-fabric** [comment]
Thanks you for approving. After this approving, what should we supposed to do?
I mean, how we can add our project to the laboratory?

**Comment by tkuhrt** [comment]
> Thanks you for approving. After this approving, what should we supposed to do? I mean, how we can add our project to the laboratory?

@sunyar-fabric : Have you resolved DCO issues in the repository yet?

**Comment by sunyar-fabric** [comment]
Yes, I did. I was using a command like this.
git rebase -i HEAD~50

**Comment by tkuhrt** [comment]
> Yes, I did. I was using a command like this. git rebase -i HEAD~50

In looking at the repo, I see a number of commits (including the latest) that do not have DCO sign-off. A number of them say "no commit message". This will need to be fixed prior to being able to transfer the repository to the Hyperledger Labs organization. Alternatively, we can create a new repository and you can do a commit with DCO sign-off (assuming you have the sign-off from other contributors). The problem with this latter approach is that it will result in a loss of history. Please let us know if we should create a new repo for you to bring the code in or if you will clean up the DCO sign-off on all commits prior to transferring the repo.

**Comment by tkuhrt** [comment]
You may find this helpful on signing previous commits: 
https://hyperledger-indy.readthedocs.io/projects/sdk/en/latest/docs/contributors/signing-commits.html#how-to-sign-previous-commits

**Comment by sunyar-fabric** [comment]
> o a commit with DCO sign-off (assuming you have the sign-off from other contributors). The problem with this latter approach is that it will re

yes, please create a new one to solve the problem.

**Comment by tkuhrt** [comment]
@sunyar-fabric : Repo created: https://github.com/hyperledger-labs/sunyar-fabric. Please bring the code over manually.

**Comment by ZankoMohammadzadeh** [comment]
Hi Tracy,
Thank you for creating the repository of SUNYAR project in Hyperledger lab.
Is it needed to run a command of "sign off" after copying the codes of
sunyar project in new repository?

sincerely,
Zanko

**Comment by tkuhrt** [comment]
Hi, Zanko.

When you do you commit, make sure to include the `-s` option. As in:

```
git commit -s -m "Message explaining the commit."
```

You may be interested in this guide: [A Complete Guide to DCO for Open Source Developers](https://www.secondstate.io/articles/dco/)

**Comment by ZankoMohammadzadeh** [comment]
Hi Tracy,

I want to commit our codes to the newly created repository of the
sunyar-fabric project.
But I encountered an error when I wanted to push our codes to the
repository.
Actually permission is denied to our github account.
[image: image.png]

Sincerely,
Zanko Mohammadzadeh

**Comment by tkuhrt** [comment]
@ZankoMohammadzadeh : I do not see the image. It appears that the original invite to the repository was not accepted. I have sent you another invite. Please accept that and then you should be able to make changes via pull requests.

**Pull request merged**
